test/common/deployment_types.go

```diff
@@ -0,0 +1,261 @@
+package common
+
+import (
+	goctx "context"
+	"testing"
+
+	appsv1 "github.com/openshift/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Namespace struct {
+	Name     string
+	Products []Product
+}
+
+type Product struct {
+	Name             string
+	ExpectedReplicas int32
+}
+
+func TestDeploymentExpectedReplicas(t *testing.T, ctx *TestingContext) {
+
+	deployment := getDeployment()
+	for _, namespace := range deployment {
+		for _, product := range namespace.Products {
+			deployment, err := ctx.KubeClient.AppsV1().Deployments(namespace.Name).Get(product.Name, v1.GetOptions{})
+			if err != nil {
+				t.Fatalf("Deployment %s not found - %s", product.Name, err)
+			}
+			if deployment.Status.Replicas != product.ExpectedReplicas {
+				t.Fatalf("Deployment %s in namespace %s doens't match the number of expected replicas %s/%s",
+					product.Name,
+					namespace.Name,
+					string(deployment.Status.Replicas),
+					string(product.ExpectedReplicas),
+				)
+			}
+		}
+	}
+}
+
+func TestDeploymentConfigExpectedReplicas(t *testing.T, ctx *TestingContext) {
+
+	deploymentConfigNS := getDeploymentConfig()
+	for _, namespace := range deploymentConfigNS {
+		for _, product := range namespace.Products {
+
+			dc := &appsv1.DeploymentConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      product.Name,
+					Namespace: namespace.Name,
+				},
+			}
+			err := ctx.Client.Get(goctx.TODO(), k8sclient.ObjectKey{Name: product.Name, Namespace: namespace.Name}, dc)
+			if err != nil {
+				t.Fatalf("DeploymentConfig %s not found - %s", product.Name, err)
```

Comment:
Instant of failing on the first error it would be better to check all deployment config so that if there are more than one missing we can list all of them in one single run